syntax = 'proto3';

import 'Characters.proto';

message PlayerJoin {
    uint32 player_id = 1;
    CharacterData character = 2;
}

message PlayerLeave {
    uint32 player_id = 1;
}

message PlayersResponse {
    message Player {
        uint32 player_id = 1;
        CharacterData character = 2;
        float velocity_x = 3;
        float velocity_y = 4;
    }
    repeated Player players = 1;
}

message JoinRequest {
    uint32 character_id = 1;
}
